pysmt/test/omt_examples.py

```diff
@@ -0,0 +1,551 @@
+from enum import Enum, auto
+import os.path as path
+import warnings
+
+from pysmt.environment import get_env
+from pysmt.optimization.goal import Goal
+from pysmt.logics import Logic
+from pysmt.fnode import FNode
+from pysmt.shortcuts import Optimizer, GE, Int, Symbol, INT, LE, GT, REAL, Real, Equals, Times, Solver, Or, Div
+from pysmt.shortcuts import BVType, BVUGE, BVSGE, BVULE, BVSLE, BVUGT, BVSGT, BVULT, BVSLT, BVZero, BVOne, BV, Ite
+from pysmt.shortcuts import And, Plus, Minus, get_env, BOOL, Implies, SBV
+from pysmt.logics import QF_LIA, QF_LRA, QF_BV, QF_NRA, QF_NIA
+from pysmt.optimization.goal import MaximizationGoal, MinimizationGoal, \
+    MinMaxGoal, MaxMinGoal, MaxSMTGoal
+from pysmt.smtlib.script import InterpreterOMT
+from pysmt.smtlib.parser.parser import SmtLib20Parser
+from pysmt.smtlib.commands import ASSERT, CHECK_SAT, CHECK_ALLSAT, MAXIMIZE, MINIMIZE
+from pysmt.exceptions import PysmtUnboundedOptimizationError, PysmtInfinitesimalError
+
+
+test_folder = path.dirname(path.abspath(__file__))
+omt_smt2_files_folder = path.join(test_folder, "smtlib", "omt")
+
+class OptimizationTypes(Enum):
+    BASIC = auto()
+    LEXICOGRAPHIC = auto()
+    PARETO = auto()
+    BOXED = auto()
+
+
+class OMTTestCase:
+    """
+    This class stores the info to test the OMT optimization methods over a precise
+    set of assertions, set of goals and the expected optimal values for the goals.
+
+    The test case is defined by the following parameters:
+    - name: the name of the test case
+    - assertions: the assertions to be added to the optimizer
+    - logic: the logic to be used by the optimizer
+    - solvable: True if the test case is solvable, False otherwise
+    - goals: a dictionary with the goals and the expected values for the goals.
+
+    The keys of the dictionary are tuples with the goals and the OptimizationType
+    corresponding to that value.
+    The values of the dictionary depends on the OptimizationType:
+    - BASIC: a list/tuple with only one value, the expected
+        value for the goal
+    - LEXICOGRAPHIC/BOXED: a list/tuple with as many values as the goals specified
+    - PARETO: a list/tuple, where each element is a list/tuple containing as many
+        values as the goals specified
+
+    The expected values of the goals can be either a FNode or a string.
+    If the expected value is a FNode, it must be a constant value.
+    If the expected value is a string, it must be either "unbounded" or "infinitesimal".
+    """
+    def __init__(self, name, assertions, logic, solvable, goals):
+        self._name = name
+        self._assertions = assertions
+        self._logic = logic
+        self._solvable = solvable
+        self._goals = goals
+
+        # consistency checks
+        assert bool(goals) == solvable, "Goals must be specified if and only if the test case is solvable"
```

Comment:
These (including all the ones below this line) should not be assertions, but soething like `PySMTValueError`s

Reply:
Ok, I put those as assert because I was not expecting the user to use this class, but only test-developers.

Setting them as exceptions anyway